src/audio.cpp

```diff
@@ -39,62 +39,71 @@ opus_stream_config_t stream_configs[MAX_STREAM_CONFIG] {
     1,
     1,
     platf::speaker::map_stereo,
+    96000,
+  },
+  {
+    SAMPLE_RATE,
+    2,
+    1,
+    1,
+    platf::speaker::map_stereo,
+    512000,
   },
   {
     SAMPLE_RATE,
     6,
     4,
     2,
     platf::speaker::map_surround51,
+    256000,
   },
   {
     SAMPLE_RATE,
     6,
     6,
     0,
     platf::speaker::map_surround51,
+    1536000,
   },
   {
     SAMPLE_RATE,
     8,
     5,
     3,
     platf::speaker::map_surround71,
+    450000,
   },
   {
     SAMPLE_RATE,
     8,
     8,
     0,
     platf::speaker::map_surround71,
+    2048000,
   },
 };
 
 auto control_shared = safe::make_shared<audio_ctx_t>(start_audio_control, stop_audio_control);
 
 void encodeThread(sample_queue_t samples, config_t config, void *channel_data) {
   auto packets = mail::man->queue<packet_t>(mail::audio_packets);
-
-  // FIXME: Pick correct opus_stream_config_t based on config.channels
-  auto stream = &stream_configs[map_stream(config.channels, config.flags[config_t::HIGH_QUALITY])];
+  auto stream  = &stream_configs[map_stream(config.channels, config.flags[config_t::HIGH_QUALITY])];
 
   opus_t opus { opus_multistream_encoder_create(
     stream->sampleRate,
     stream->channelCount,
     stream->streams,
     stream->coupledStreams,
     stream->mapping,
-    OPUS_APPLICATION_AUDIO,
+    OPUS_APPLICATION_RESTRICTED_LOWDELAY,
     nullptr) };
 
-  // For some reason, audio is crackling when the encoder is set to constant bitstream.
-  // We simulate a constant bitstream with OPUS_SET_BITRATE(OPUS_BITRATE_MAX) -->
-  // which tries to occupy as much space as possible in the packet
```

Comment:
I tested this on hosts running Windows 11 and Fedora 37 and didn't note any crackling, so whatever bug was causing the crackling with CBR mode seems to have been fixed.

Reply:
I'm guessing there were fixes upstream that resolved this.